pkg/arvo/app/chat-cli.hoon

```diff
@@ -0,0 +1,1111 @@
+::
+|%
++$  state
+  $:  grams=(list mail)                             ::  all messages
+      known=(set [target serial])                   ::  known message lookup
+      count=@ud                                     ::  (lent grams)
+      bound=(map target char)                       ::  bound circle glyphs
+      binds=(jug char target)                       ::  circle glyph lookup
+      audience=(set target)                         ::  active targets
+      settings=(set term)                           ::  frontend flags
+      width=@ud                                     ::  display width
+      timez=(pair ? @ud)                            ::  timezone adjustment
+      cli=[=bone state=sole-share:sole-sur]         ::  console id & state
+  ==
+::
++$  mail  [source=target envelope]
++$  target  [=ship =path]
+::
++$  glyph  char
+++  glyphs  "!@#$%^&()-=_+[]\{};'\\:\",.<>?"
+::
++$  command
+  $%  [%target (set target)]                        ::  set messaging target
+      [%say (list letter)]                          ::  send message
+      [%eval cord hoon]                             ::  send #-message
+    ::
+      [%create path =(unit glyph)]                  ::  create chat
+      [%join target =(unit glyph)]                  ::  join target
+      [%leave target]                               ::  nuke target
+    ::
+      [%bind glyph target]                          ::  bind glyph
+      [%unbind glyph]                               ::  unbind glyph
+      [%what (unit $@(char target))]                ::  glyph lookup
+    ::
+      [%settings ~]                                 ::  show active settings
+      [%set term]                                   ::  set settings flag
+      [%unset term]                                 ::  unset settings flag
+      [%width @ud]                                  ::  adjust display width
+      [%timezone ? @ud]                             ::  adjust time printing
+    ::
+      [%select $@(rel=@ud [zeros=@u abs=@ud])]      ::  rel/abs msg selection
+      [%chats ~]                                    ::  list available chats
+      [%help ~]                                     ::  print usage info
+  ==                                                ::
+::
++$  move  [bone card]
++$  card
+  $%  [%diff %sole-effect sole-effect:sole-sur]
+      [%poke wire dock out-action]
+      [%peer wire dock path]
+  ==
+::
++$  out-action
+  $%  [%chat-action chat-action]
+      [%chat-view-action chat-view-action]
+      [%chat-hook-action chat-hook-action]
+  ==
+--
+::
+|_  [=bowl:gall state]
+++  this  .
+::  +prep: setup & state adapter
+::
+++  prep
+  |=  old=(unit state)
+  ~&  %chat-cli-prep
+  ?^  old
+    [~ this(+<+ u.old)]
+  :-  [connect ~]
+  %_  this
+    audience  [[our-self /] ~ ~]
+    settings  (sy %showtime %notify ~)
+    width  80
+  ==
+::  +connect: connect to the chat-store
+::
+++  connect
+  ^-  move
+  [ost.bowl %peer /chat-store [our-self %chat-store] /all]
```

Comment:
Could you use /updates instead and scry when you need access to larger chunks of state?

Reply:
Since we'll be starting this on-boot, the `/all` here is essentially equivalent to `/updates`, though I do indeed see issues around pulling in all of everything if we ever need to re-subscribe because this dropped.

But then, if I manually re-sub after this connection breaking, then I do want to see what I missed, ideally without getting all of everything again. Hall allowed you to say "here's the last timestamp I've seen, show me everything that happened from that point on". Does chat-store have an equivalent affordance for catching up?

Reply:
It has peek arms that implement that functionality, but not peer arms.

Reply:
> though I do indeed see issues around pulling in all of everything if we ever need to re-subscribe because this dropped.

This'll be a re-subscribe to a local `chat-store`, right?  If so, then even if it's big it should be fast as long as you're just passing it around -- it's just a pointer, after all.